client/client-multi/client-cpp/impl/KaaClient.cpp

```diff
@@ -38,21 +38,19 @@
 
 #include "kaa/logging/Log.hpp"
 #include "kaa/context/SimpleExecutorContext.hpp"
-#include "kaa/DummyKaaClientStateListener.hpp"
 #include "kaa/KaaClientProperties.hpp"
 #include "kaa/logging/DefaultLogger.hpp"
 
 namespace kaa {
 
-KaaClient::KaaClient(IKaaClientPlatformContextPtr context, IKaaClientStateListenerPtr listener)
-    : logger_(new DefaultLogger(context->getProperties().getClientId(), context->getProperties().getLogFileName())),
-      context_(context->getProperties(), *logger_, context->getExecutorContext()),
+KaaClient::KaaClient(IKaaClientPlatformContextPtr platformContext, IKaaClientStateListenerPtr listener)
+    : logger_(new DefaultLogger(platformContext->getProperties().getClientId(), platformContext->getProperties().getLogFileName())),
+      context_(platformContext->getProperties(), *logger_, platformContext->getExecutorContext(), nullptr, listener),
       status_(new ClientStatus(context_)),
-      stateListener_(listener),
-      platformContext_(context)
+      platformContext_(platformContext)
 {
-    if (!stateListener_) {
-        stateListener_ = std::make_shared<DummyKaaClientStateListener>();
+    if (context_.getClientStateListener() == nullptr) {
+        throw KaaException("State listener is null");
```

Comment:
Listener is not mandatory! I used a dummy implementation just to not to check for null in other places.